test/test_linalg.py

```diff
@@ -2620,6 +2620,202 @@ def test_old_matrix_rank(self, device, dtype):
             self.assertEqual(torch.matrix_rank(aaT, True), np.linalg.matrix_rank(aaT.cpu().numpy(), True))
             self.assertEqual(torch.matrix_rank(aaT, 0.01, True), np.linalg.matrix_rank(aaT.cpu().numpy(), 0.01, True))
 
+    @dtypes(torch.double)
```

Comment:
Can we have complex dtype be tested here as well?
```suggestion
    @dtypes(torch.double, torch.cdouble)
```